docs/website/scripts/live-blocks/src/constants.js

```diff
@@ -0,0 +1,188 @@
+  BLOCK_CONTAINER_MERGED_DOWN: join(BLOCK_CONTAINER, 'merged-down'),
+  BLOCK_CONTAINER_ERROR: join(BLOCK_CONTAINER, 'error'),
+  HUGO_BLOCK_CONTAINER: 'highlight',
+  ICON_BAR_OUTER: join(ICON_BAR, 'outer'),
+  ICON_BAR_INNER: join(ICON_BAR, 'inner'),
+  ICON_CONTAINER,
+  ICON_CONTAINER_INDICATOR: join(ICON_CONTAINER, 'indicator'),
+  ICON_CONTAINER_BUTTON: join(ICON_CONTAINER, 'button'),
+  ICON_IMG,
+  ICON_IMG_SPINNING: join(ICON_IMG, 'spinning')
+}
+
+// --- LABELS ---
+// prefix for "lang" labels for live code blocks
+export const LABEL_PREFIX = 'live:'
+
+// The types of live code blocks with their names as found in labels
+export const BLOCK_TYPES = {
+  MODULE: 'module',
+  QUERY: 'query',
+  INPUT: 'input',
+  OUTPUT: 'output',
+}
+
+// The types of expected error tags that output blocks can have with their names as found in labels
+export const EXPECTED_ERROR_TAG_TYPES = {
+  EXPECT_ERROR: 'expect_error',
+  EXPECT_UNDEFINED: 'expect_undefined',
+  // from 'ast'
+  EXPECT_REGO_ERROR: 'expect_rego_error',
+  EXPECT_PARSE_ERROR: 'expect_parse_error',
+  EXPECT_COMPILE_ERROR: 'expect_compile_error',
+  EXPECT_REGO_TYPE_ERROR: 'expect_rego_type_error',
+  EXPECT_UNSAFE_VAR: 'expect_unsafe_var',
+  EXPECT_RECURSION: 'expect_recursion',
+  // from 'topdown'
+  EXPECT_EVAL_ERROR: 'expect_eval_error',
+  EXPECT_CONFLICT: 'expect_conflict',
+  EXPECT_EVAL_TYPE_ERROR: 'expect_eval_type_error',
+  EXPECT_BUILTIN_ERROR: 'expect_builtin_error',
+  EXPECT_WITH_MERGE_ERROR: 'expect_with_merge_error',
+  // more specific
+  EXPECT_ASSIGNED_ABOVE: 'expect_assigned_above',
+  EXPECT_REFERENCED_ABOVE: 'expect_referenced_above',
+}
+
+// The types of tags that blocks can have with their names as found in labels
+export const TAG_TYPES = {
+  HIDDEN: 'hidden',
+  READ_ONLY: 'read_only',
+  MERGE_DOWN: 'merge_down',
+  OPENABLE: 'openable',
+  ...EXPECTED_ERROR_TAG_TYPES,
+}
+
+// Regular expressions for breaking apart labels, contructing them this way makes it easier to only accept valid ones while only specifying them in one place.
+const BLOCK_REG_EXP = Object.values(BLOCK_TYPES).join('|')
+const TAG_REG_EXP = Object.values(TAG_TYPES).join('|')
+export const LABEL_REG_EXP = new RegExp(`^${LABEL_PREFIX}(\\w+(?:\\/\\w+)*):(${BLOCK_REG_EXP})(:(?:(?:${TAG_REG_EXP}),)*(?:${TAG_REG_EXP})|)$`)
+
+// CSS-style selector for finding the live code blocks in a document
+export const BLOCK_SELECTOR = `code[data-lang^="${LABEL_PREFIX}"]`
+
+// The name for the module file that gets evaluated (specifically shows up in error messages.)
+export const EVAL_MODULE_NAME = 'module.rego'
+
+// --- PREPROCESSING ---
+// The maximum number of files to be processing at any given time. 0 indicates unlimited.
+export const MAX_CONCUR_FILES = 0
+// The maximum number of evaluations to be performing for a single processing file at any given time (net maximum is MAX_CONCUR_FILES * MAX_CONCUR_FILE_EVALS). 0 indicates unlimited.
+export const MAX_CONCUR_FILE_EVALS = 0
```

Comment:
Instead of defaulting to unlimited can we default to the number of cpu cores available https://nodejs.org/api/os.html#os_os_cpus ? It is generally not going to be any faster if we process any more than that in parallel, and it helps avoid swamping small systems trying to do too many at the same time.

Reply:
Hmm, From my understanding NodeJS is just running in a single thread and some of the delays are going to be FS-related, for those cases limiting based on CPUs doesn't really make sense. The only place where multiple cores could be used is the OPA evals. Maybe I'll try processing a bunch of files and try to peg all my cores, I'm _guessing_ I won't be able to. Is there some other metric that might make sense to place limits with (I didn't know you could get the # of cores), pegging disk throughput probably isn't a good idea either :)

Reply:
Ohh.. hah, yea, thats a good point. I also forgot one of my comments which was that we could use probably use worker threads to speed this up if we needed to.

Reply:
Definitely; I think it might take some refactoring around where the opa binary acquiring process happens but really shouldn't be too bad if we want to add it later (or now, I'll see what times look like with some more files).

Reply:
@patrick-east I just did a bunch of testing, I can't get get it to process any faster than when I do 1 file and a few tens of evals at a time (~1.2s per how-do-i-write), although not limiting files goes _way_ slower (~3x). It seems to self-limit to about 40% CPU usage so I don't think tighter rate limits are going to be helpful and the usage is pretty even across cores so I don't think worker threads will do much either (looks like OPA eval is the brunt of the work).